src/js/Content/Features/Community/Badges/FBadgeSortAndFilter.js

```diff
@@ -1,65 +1,50 @@
 import {HTML, Localization} from "../../../../modulesCore";
-import {CommunityUtils, DOMHelper, Feature} from "../../../modulesContent";
+import {Feature} from "../../../modulesContent";
 import {Page} from "../../Page";
 
 export default class FBadgeSortAndFilter extends Feature {
 
     apply() {
+
+        this._header = document.querySelector(".profile_badges_header");
+
         this._addSort();
         this._addFilter();
         this._addViewOptions();
+
+        Page.runInPageContext(() => { window.SteamFacade.bindAutoFlyoutEvents(); });
     }
 
     _addSort() {
 
-        const isOwnProfile = CommunityUtils.currentUserIsOwner();
-        const sorts = ["c", "a", "r"];
-
-        const sorted = document.querySelector("a.badge_sort_option.active").search.replace("?sort=", "")
-            || (isOwnProfile ? "p" : "c");
-
-        let linksHtml = "";
-
-        if (isOwnProfile) {
-            sorts.unshift("p");
-        }
+        const isOwnProfile = this.context.myProfile;
+        const sortOptions = this._header.querySelector(".profile_badges_sortoptions");
 
-        // Build dropdown links HTML
-        const nodes = document.querySelectorAll(".profile_badges_sortoptions a");
-        let i = 0;
-        for (const node of nodes) {
-            linksHtml += `<a class="popup_menu_item by_${sorts[i]}" data-sort-by="${sorts[i]}" href="?sort=${sorts[i]}">${node.textContent.trim()}</a>`;
-            i++;
+        // Build popup menu for Steam's sort options + our options
+        let html = "";
+        for (const node of sortOptions.querySelectorAll("a")) {
+            const sort = new URL(node.href).searchParams.get("sort");
+            html += `<a class="popup_menu_item" href="?sort=${sort}">${node.textContent.trim()}</a>`;
         }
         if (isOwnProfile) {
-            linksHtml += `<a class="popup_menu_item by_d" data-sort-by="d" id="es_badge_sort_drops">${Localization.str.most_drops}</a>`;
-            linksHtml += `<a class="popup_menu_item by_v" data-sort-by="v" id="es_badge_sort_value">${Localization.str.drops_value}</a>`;
+            html += `<a class="popup_menu_item" id="es_badge_sort_drops">${Localization.str.most_drops}</a>`;
+            html += `<a class="popup_menu_item" id="es_badge_sort_value">${Localization.str.drops_value}</a>`;
         }
 
-        const container = document.createElement("span");
-        container.id = "wishlist_sort_options";
-        DOMHelper.wrap(container, document.querySelector(".profile_badges_sortoptions"));
+        const activeText = sortOptions.querySelector("a.active").textContent.trim();
```

Comment:
Same here, do we need the `a` in the selector?

Reply:
Oh I changed this before limiting the selection to a descendant of `sortOptions`, shouldn't hurt either right?